api/v1/pooler_types.go

```diff
@@ -80,6 +80,14 @@ type PoolerSpec struct {
 	// The configuration of the monitoring infrastructure of this pooler.
 	// +optional
 	Monitoring *PoolerMonitoringConfiguration `json:"monitoring,omitempty"`
+
+	// What type of service to create for the pooler. Default: ClusterIP.
+	// +optional
+	ServiceType corev1.ServiceType `json:"servicetype,omitempty"`
```

Comment:
Ideally this could fall under a serviceTemplate, providing a full service spec and metadata for labels and annotations, we could still apply the labels/annotation inheritance, but allow setting service specific ones here.

Reply:
I agree